elements/includes/creative_commons.inc

```diff
@@ -72,269 +124,941 @@ function xml_form_elements_creative_commons($element, &$form_state) {
-    $properties = explode('-', array_pop($default_value_array));
+  // Get current values, process them and store them in the element.
+  $input = isset($form_state['input'][$element['#name']]) ? $form_state['input'][$element['#name']] : NULL;
+  $values = xml_form_elements_creative_commons_process_input($element, $input, $form_state);
+  $element['#value'] = $values;
+  // Force values into $form_state['input']
+  $form_state['input'][$element['#name']] = $values;
 
-    $commercial = 'y';
-    $derivatives = 'y';
-
-    foreach ($properties as $property) {
-      switch ($property) {
-        case 'by':
-          break;
-
-        case 'nc':
-          $commercial = 'n';
-          break;
-
-        case 'nd':
-          $derivatives = 'n';
-          break;
-
-        case 'sa':
-          $derivatives = 'sa';
-          break;
-      }
-    }
-
-    $jurisdiction = empty($default_value_array[1]) ? $default_jurisdiction : $default_value_array[1];
-
-  }
-  else {
-    $derivatives = isset($form_state['input'][$element['#name']]['license_fieldset']['allow_modifications']) ? $form_state['input'][$element['#name']]['license_fieldset']['allow_modifications'] : 'y';
-    $commercial = isset($form_state['input'][$element['#name']]['license_fieldset']['allow_commercial']) ? $form_state['input'][$element['#name']]['license_fieldset']['allow_commercial'] : 'y';
-    $jurisdiction = isset($form_state['input'][$element['#name']]['license_fieldset']['license_jurisdiction']) ? $form_state['input'][$element['#name']]['license_fieldset']['license_jurisdiction'] : $default_jurisdiction;
-    if (!isset($form_state['input'][$element['#name']]['license_fieldset']['disabled']) &&
-      isset($form_state['input'][$element['#name']]['license_fieldset'])) {
-      // If the disabled element is missing but the license fieldset exists.
-      $disabled = FALSE;
-    }
-  }
+  // Store the license type options in a variable.
+  $license_options = xml_form_elements_creative_commons_license_type_options();
 
   // Form elements.
+
   $element['license_fieldset'] = array(
+    '#id' => $license_fieldset_id,
     '#type' => 'fieldset',
     '#collapsed' => FALSE,
     '#collapsible' => TRUE,
     '#title' => t('License'),
   );
 
-  $element['license_fieldset']['disabled'] = array(
-    '#type' => 'checkbox',
-    '#title' => t("Don't output Creative Commons license."),
-    '#description' => t('Check this box to avoid outputting a creative commons license. This will still output a blank element, so make sure to use a cleanup template to remove it.'),
-    '#default_value' => $disabled,
+  $element['license_fieldset']['license_type'] = array(
+    '#type' => 'select',
+    '#title' => t('Select a license type'),
+    '#description' => t('Select "@option" to avoid outputting a creative commons license. This will still output a blank element in the resulting XML, so make sure to use a cleanup template to remove it.', array('@option' => $license_options['none'])),
```

Comment:
This could appear only if "none" is selected, drupal's ajax re-renders the element on background, so could accomplished. Not a big deal so only if possible. "This will still output a blank element in the resulting XML, so make sure to use a cleanup template to remove it" things like this make sense if the form user is the admin and understands dark logic of islandora. Same message will freak out any metadata user that is not part of this religion.

Reply:
Oops! I had misread the requirement. Now it should be o.k..